### PR TITLE
rename node_edges to incident_edges & make updating n_edges more consistent

### PR DESCRIPTION
**Pull request by jwindhager**
for consistency, update _n_edges in concrete implementations only

also, rename _remove_node_edges to _remove_incident_edges

### CODE REVIEW

File: src/napari_graph/_base_graph.py [hunk 1]
```diff
@@ -546,6 +546,18 @@ def _validate_edges(self, edges: ArrayLike) -> np.ndarray:
 
         return edges
 
+    @abstractmethod
+    def _add_edge(self, src_node: int, tgt_node: int) -> None:
+        raise NotImplementedError
+
+    def add_edge(self, src_node: int, tgt_node: int) -> None:
```
Comment (JoOkuma) [inline src/napari_graph/_base_graph.py]: To reduce duplicated code I would remove the `_add_edge` method and `add_edge` would only wrap a `add_edges` call, for example:

```python3
def add_edge(see, src_node: int, tgt_node: int) -> None:
    self.add_edges([[src_node, tgt_node]])
```

Since it's a single node that's being inserted I don't think performance is critical, otherwise, it would be using the `add_edges` method.

The same is valid for `remove_edge`.

**Comment by JoOkuma** [comment]
I agree with the other improvements, but I'm not sure that will guarantee the buffers and lengths are correct when an error is raised because when an error is raised in `_remove_edges` it won't return any value, thus `n_edges` won't be updated.

What do you think of changing `remove_edges` to something like this:
```python3
    def remove_edges(self, edges: ArrayLike) -> None:
        """Remove edges from the graph.
        Parameters
        ----------
        edges : ArrayLike
            A list of 2-dimensional tuples or an Nx2 array with a pair of
            node indices.
        """
        edges = self._validate_edges(edges)
        edges = self._map_world2buffer(edges)
        try:
            self._remove_edges(edges)
        except ValueError as e:
            self._n_edges = ...   # recompute n_edges by transversing edges buffer and counting valid edges
            raise e
```

If we do this I think it's worth adding a test for when trying to remove an invalid edge.

File: src/napari_graph/_base_graph.py (hunk 1 above)
Comment (jwindhager) [reply src/napari_graph/_base_graph.py]: Looking at this again, I agree. Actually, I'm not so sure anymore whether we should actually provide functions for adding/removing single nodes/edges :see_no_evil: Thinking about it, this could potentially incentivize inefficient programming (e.g. adding nodes in a loop), and just calling the corresponding multi-item functions like in your example is probably simple enough. What do  you think? Should we keep the single-node/edge functions?

**Comment by jwindhager** [comment]
I was actually thinking about this and even had a try/finally implemented in an earlier version, where the multi-item functions returned the number of modified nodes/edges (which, without errors, should correspond to the number of nodes/edges passed to the function).

However, I *think* that what we'd like is either (a) the entire operation succeeds or (b) the entire option fails, not some arbitrary intermediate. This would require that we sufficiently check the validity of the arguments (e.g. existence of edges when deleting) prior to executing the operation. What do you think?

File: src/napari_graph/_base_graph.py (hunk 1 above)
Comment (JoOkuma) [reply src/napari_graph/_base_graph.py]: I agree with you. We can keep just the renaming.

**Comment by JoOkuma** [comment]
I agree that failing without modifying the data is better, it will make the removal operation slower, but I think that should be fine because a user will probably just remove a few edges at a time when using interactively (through napari).

File: src/napari_graph/_base_graph.py (hunk 1 above)
Comment (jwindhager) [reply src/napari_graph/_base_graph.py]: How about the `add_node`/ `remove_node` methods? Should they also be changed to `add_nodes`/`remove_nodes` for consistency?

Comment (JoOkuma) [reply src/napari_graph/_base_graph.py]: Yes, I think we could change it, I would still keep it in python (not numba), though.

Comment (jwindhager) [reply src/napari_graph/_base_graph.py]: Removed the methods again. In the interest of time, leaving the `add_node`/`remove_node` methods as they are for now.
